pkg/controllers/cassandra/nodepool/resource.go

```diff
@@ -44,6 +47,26 @@ func StatefulSetForCluster(
 				Type: apps.RollingUpdateStatefulSetStrategyType,
 			},
 			PodManagementPolicy: apps.ParallelPodManagement,
+			VolumeClaimTemplates: []apiv1.PersistentVolumeClaim{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: cassDataVolumeName,
+						Annotations: map[string]string{
+							"volume.beta.kubernetes.io/storage-class": np.Persistence.StorageClass,
+						},
+					},
+					Spec: apiv1.PersistentVolumeClaimSpec{
+						AccessModes: []apiv1.PersistentVolumeAccessMode{
+							apiv1.ReadWriteOnce,
+						},
+						Resources: apiv1.ResourceRequirements{
+							Requests: apiv1.ResourceList{
+								apiv1.ResourceStorage: np.Persistence.Size,
+							},
+						},
+					},
+				},
+			},
```

Comment:
We should only be setting the PVC template if persistence is enabled, if it's not, we should add an extra `spec.template.spec.volumes` which is an emptyDir with the name `cassDataVolumeName` so that the volumeMount references a valid volume still.